backend/market/views.py

```diff
@@ -299,5 +308,6 @@ def destroy(self, request, *args, **kwargs):
     def list(self, request, *args, **kwargs):
         if not Item.objects.filter(pk=int(self.kwargs["item_id"])).exists():
             raise exceptions.NotFound("No Item matches the given query")
-        self.check_object_permissions(request, Item.objects.get(pk=int(self.kwargs["item_id"])))
+        for offer in self.get_queryset():
```

Comment:
On another note, do you have to manually raise NotFound exception? If you do, shouldn't you be raising in the queryset?